.ci/README.md

```diff
@@ -0,0 +1,31 @@
+CI Design guidelines
+
+* It is more maintainable to create scripts in `.ci` and then call them from the workflows than to
+  have scripts inline in the workflows. However, it is also good to split up scripts in multiple
+  steps and jobs depending on what is being done.
+
+* The docker image is rebuilt if the `Dockerfile` or `.containerversion` file is modified. (In case
+  of a push event it is also automatically published to docker hub).
```

Comment:
What happens in the PR CI that modifies the Dockerfile? Technically the CI should run all commits starting from the one that modifies the Dockerfile against the new image, and the ones before against the old image.

Often Dockerfile modifications require other changes too, like fixing new clippy linters when updating the Rust toolchain, etc, and one wants to see if the PR passes CI with the new image.

Reply:
Yes, this is exactly how it works. See my test here: https://github.com/NickeZ/bitbox02-firmware/pull/3

Reply:
Amazing!